src/linear_proof.rs

```diff
@@ -164,10 +175,24 @@ impl LinearProof {
         // Public scalar vector b
         b_vec: Vec<Scalar>,
     ) -> Result<(), ProofError> {
+        let n = b_vec.len();
+        if G.len() != n {
+            return Err(ProofError::InvalidGeneratorsLength);
+        }
+
+        // Append all public data to the transcript
         transcript.innerproduct_domain_sep(n as u64);
         transcript.append_point(b"C", &C);
-        let (x_vec, x_inv_vec, b_0) = self.verification_scalars(n, transcript, b_vec)?;
+        for b_i in &b_vec {
+            transcript.append_scalar(b"b_i", b_i);
```

Comment:
Are we adding the $b_i$ into the transcript twice? It looked like it was added above around line ~70ish.